src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/PersonBlockingRulesRepository.kt

```diff
@@ -11,10 +14,29 @@ class PersonBlockingRulesRepository {
   @PersistenceContext
   private val entityManager: EntityManager? = null
 
-  fun findMatchCandidates(searchCriteria: PersonSearchCriteria, personQuery: String): List<PersonEntity> {
+  fun findMatchCandidates(searchCriteria: PersonSearchCriteria, personQuery: String, pageable: Pageable): Page<PersonEntity> {
     val query = entityManager!!.createNativeQuery(personQuery, PersonEntity::class.java)
     query.setParameter("firstName", searchCriteria.firstName)
     query.setParameter("lastName", searchCriteria.lastName)
-    return query.resultList as List<PersonEntity>
+
+    // apply pagination
+    query.setFirstResult((pageable.offset.toInt()))
+    query.maxResults = pageable.pageSize
+
+    val results = query.resultList as List<PersonEntity>
+
+    val countQuery = countMatchCandidates(personQuery, searchCriteria)
```

Comment:
I think this wants to be outside of the pageable scope first, otherwise this we be called multiple times increasing processing time